Adds giotto-tda to "Related projects" (#18427)
scikit-learn/scikit-learn: doc/related_projects.rst

  Feature-engine allows the application of preprocessing steps to selected groups
  of variables and it is fully compatible with the Scikit-learn Pipeline.

**Topological Data Analysis**

> - `giotto-tda <https://github.com/giotto-ai/giotto-tda>`_ A library for `Topological Data Analysis

Reviewer: I know we are not consistent with `rst` file, but can this be updated to use < 80 characters per line?

Author: Thanks for the review! This is now done.

  Feature-engine allows the application of preprocessing steps to selected groups
  of variables and it is fully compatible with the Scikit-learn Pipeline.

**Topological Data Analysis**

> - `giotto-tda <https://github.com/giotto-ai/giotto-tda>`_ A library collecting
>   state-of-the-art implementations of several algorithms in `Topological Data Analysis
>   <https://en.wikipedia.org/wiki/Topological_data_analysis>`_. It offers a
>   scikit-learn–compatible API, a suite of tools to transform various kinds of data
>   inputs (point clouds, graphs, time-series, images, etc.) into forms suitable for
>   computations of topological summaries, and components dedicated to extracting rich 
>   sets of scalar features of topological origin, which can be used alongside other feature
>   extraction methods in scikit-learn and/or in the context of complex scikit-learn pipelines.

Reviewer 2: Thanks @ulupo ! Generally +1 for inclusion, but I think it would be good to make the description a bit shorter and more neutral, similarly to other entries there.

I have tried to shorten it below, but of course feel free to change the wording differently if you prefer

```suggestion
- `giotto-tda <https://github.com/giotto-ai/giotto-tda>`_ A library for `Topological Data Analysis
  <https://en.wikipedia.org/wiki/Topological_data_analysis>`_ aiming to provide a
  scikit-learn compatible API. It offers tools to transform data
  inputs (point clouds, graphs, time-series, images) into forms suitable for
  computations of topological summaries, and components dedicated to extracting
  sets of scalar features of topological origin, which can be used alongside other feature
  extraction methods in scikit-learn.
```

**Note to reviewers:** I have done a minor contribution to this package and collaborated with authors in the past, so we would need a second independent review )
